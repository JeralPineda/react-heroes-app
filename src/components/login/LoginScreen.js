import React, { useContext } from 'react';
import { AuthContext } from '../../auth/AuthContext';
import { types } from '../../types/types';

import './login.css';

export const LoginScreen = ({ history }) => {
   const { dispatch } = useContext(AuthContext);

   const handleLogin = () => {
      // Navegamos a otra pantalla
      //   history.push('/');

      //   Disparar acci√≥n
      dispatch({
         type: types.login,
         payload: {
            name: 'Jeral',
         },
      });

      history.replace('/'); //remplaza en la historia
   };

   return (
      <div className='container p-5'>
         <div className='row rounded-lg ali'>
            <div className='col-md-4 mx-auto'>
               <div className='card fondo text-center color'>
                  <div className='card-header'>
                     <h3>Heroes App</h3>
                  </div>
                  <img src='./logo192.png' alt='Logo App' className='card-img-top mx-auto m-2 rounded-circle w-50' />
                  <div className='card-body'>
                     <div className='d-flex justify-content-center mb-2'>
                        <button onClick={handleLogin} className='btn btn-secondary'>
                           LogIn
                        </button>
                     </div>
                  </div>
               </div>
            </div>
         </div>
      </div>
   );
};
