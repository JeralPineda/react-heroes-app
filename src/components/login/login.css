body {
   background-color: rgb(224, 226, 235);
   color: #222;
}

.container {
   margin-top: 9rem !important;
}

.card {
   background-color: rgb(197, 199, 206);
   color: #222;
   border-color: rgb(197, 199, 206);
   border-radius: 5px !important;
   box-shadow: 0 8px 18px -8px rgba(19, 20, 24, 0.973);
}

.card-body a {
   text-decoration: none;
   color: #222;
}

.card-body a:hover {
   color: #222;
   width: 5px;
}
