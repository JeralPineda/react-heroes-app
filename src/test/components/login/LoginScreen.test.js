import { mount } from 'enzyme';
import React from 'react';
import { AuthContext } from '../../../auth/AuthContext';
import { LoginScreen } from '../../../components/login/LoginScreen';
import { types } from '../../../types/types';

describe('Pruebas en <LoginScreen />', () => {
   const history = {
      replace: jest.fn(),
   };

   const contextValue = {
      dispatch: jest.fn(),
      user: {
         logged: true,
      },
   };

   const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
         <LoginScreen history={history} />
      </AuthContext.Provider>
   );

   test('debe de mostrarse correctamente', () => {
      expect(wrapper).toMatchSnapshot();
   });

   test('debe de realizar el dispatch y la navegaciÃ³n', () => {
      const handleClick = wrapper.find('button').prop('onClick');

      handleClick();
      expect(contextValue.dispatch).toHaveBeenCalledWith({
         type: types.login,
         payload: {
            name: 'Jeral',
         },
      });

      expect(history.replace).toHaveBeenCalledWith('/');

      //   Simular que grabamos algo en localstorage
      localStorage.setItem('lastPath', '/dc');
      handleClick();
      expect(history.replace).toHaveBeenCalledWith('/dc');
   });
});
