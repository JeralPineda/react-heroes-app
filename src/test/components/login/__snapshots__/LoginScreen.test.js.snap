// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <LoginScreen /> debe de mostrarse correctamente 1`] = `
<div
  className="container p-5"
>
  <div
    className="row rounded-lg ali"
  >
    <div
      className="col-md-4 mx-auto"
    >
      <div
        className="card fondo text-center color"
      >
        <div
          className="card-header"
        >
          <h3>
            Heroes App
          </h3>
        </div>
        <img
          alt="Logo App"
          className="card-img-top mx-auto m-2 rounded-circle w-50"
          src="./logo192.png"
        />
        <div
          className="card-body"
        >
          <div
            className="d-flex justify-content-center mb-2"
          >
            <button
              className="btn btn-secondary"
              onClick={[Function]}
            >
              LogIn
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
